appsec/src/extension/tags.c

```diff
@@ -1423,6 +1433,110 @@ static PHP_FUNCTION(datadog_appsec_track_user_login_failure_event)
     dd_tags_set_sampling_priority();
 }
 
+static PHP_FUNCTION(datadog_appsec_track_authenticated_user_event_automated)
+{
+    UNUSED(return_value);
+    if (!DDAPPSEC_G(active)) {
+        mlog(dd_log_debug, "Trying to access to track_user_login_failure_event "
+                           "function while appsec is disabled");
+        return;
+    }
+
+    zend_string *user_id;
+    zend_string *anon_user_id = NULL;
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "S", &user_id) == FAILURE) {
+        mlog(dd_log_warning,
+            "Unexpected parameter combination, expected user_id");
+        return;
+    }
+
+    if (ZSTR_LEN(user_id) == 0) {
+        mlog(dd_log_warning, "Unexpected empty user id");
+        return;
+    }
+
+    zval *nullable meta = _root_span_get_meta();
+    if (!meta) {
+        return;
+    }
+
+    _user_event_triggered = true;
+    zend_array *meta_ht = Z_ARRVAL_P(meta);
+
+    user_collection_mode mode = dd_get_user_collection_mode();
+    if (mode == user_mode_disabled ||
+        !get_DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING_ENABLED()) {
+        return;
+    }
+
+    if (mode == user_mode_anon) {
+        anon_user_id = dd_user_info_anonymize(user_id);
+        if (!anon_user_id) {
+            mlog(dd_log_debug, "Failed to anonymize user ID");
+            return;
+        }
+    }
+
+    // usr.id = <user_id>
+    _add_new_zstr_to_meta(meta_ht, _dd_tag_user_id,
+        anon_user_id ? anon_user_id : user_id, true, false);
+
+    // _dd.appsec.usr.id = <user_id>
+    _add_new_zstr_to_meta(meta_ht, _dd_appsec_user_id,
+        anon_user_id ? anon_user_id : user_id, !anon_user_id, true);
+
+    // _dd.appsec.user.collection_mode =
+    // <DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING>
+    _add_new_zstr_to_meta(meta_ht, _dd_user_collection_mode,
+        dd_get_user_collection_mode_zstr(), true, false);
+
+    dd_tags_set_sampling_priority();
+}
+
+static PHP_FUNCTION(datadog_appsec_track_authenticated_user_event)
```

Comment:
Shouldn't these two functions result in libddwaf calls?

Reply:
Yes they do, added them